### ISSUE TITLE
re. StackNavigator swipe back gesture conflict with the ScrollView scroll gesture on IOS

### ISSUE THREAD

**Issue by cokepizza** [react-navigation/react-navigation#8946]
Current Behavior

I changed the Stack Navigation setting so that I can swipe back anywhere on the screen.
gestureResponseDistance.horizontal: screenWidth

After that, The swipe back gesture of stackNavigator and the gesture of scrollView collide.
I've looked at several similar cases in the issue list here and tried and failed.

I tried using scrollView in react-native-gesture-handler, but nothing worked.
I tried using GestureHandlerRefContext and waitFor property, but nothing worked.

import { NativeViewGestureHandler } from 'react-native-gesture-handler';
import { GestureHandlerRefContext } from '@react-navigation/stack';
import { ScrollView } from 'react-native-gesture-handler';

<GestureHandlerRefContext.Consumer>
                {ref => (
                    <NativeViewGestureHandler waitFor={ref}>
                        <ScrollView
                            keyExtractor={(item, index) => `image_${item}`}
                            horizontal
                            initialScrollIndex={0}
                            pagingEnabled={true}
                            showsHorizontalScrollIndicator={false}
                            scrollEventThrottle={8}
                            onScroll={event => {
                                onScroll({
                                    index: Math.round(event.nativeEvent.contentOffset.x / screenWidth),
                                });
                            }}
                        >
                    
                            {feed.images.map((image, index) => (
                                <Image key={`feedImage_${index}`} uri={image} />
                            ))}
                        </ScrollView>
                    </NativeViewGestureHandler>
                )} 
</GestureHandlerRefContext.Consumer>
Expected Behavior

I want the scrollView to always receive the gesture event first and only have the gesture control in the stackNavigation when the scrollView's index is 0.
In other words, I want only scrollView's gesture event to work inside scrollView. If there is an exception, I want the swipe back of the stack navigation to work specifically only when the index is 0 in the scrollView.
Of course, Outside of the scrollView, the stackNavigation gesture always has control, so you should be able to swipe back anywhere on the screen.

How to reproduce

Your Environment

"@react-navigation/native": "^5.7.3",
"@react-navigation/stack": "^5.9.0",
"react-native": "0.61.5",
"react-native-gesture-handler": "^1.8.0",

**Comment by cokepizza** [comment]
I haven't solved the problem yet. Can I get some help?

**Comment by MoOx** [comment]
Dup of #8813

**Comment by SKempin** [comment]
@cokepizza @MoOx I also have this issue, has there been a fix?

**Comment by pavelglebov** [comment]
Any action on this?

**Comment by githuboftigran** [comment]
Facing the same issue. Any updates?

**Comment by Aryk** [comment]
Me too, the best I could come up with...is shift your scroll view over by 15 pixels (and adjust the items)...but that's super hacky...

**Comment by rohankm** [comment]
any fix??
